### PR TITLE
Add docs for new authentication

### LINKED ISSUE #3008: Document new authentication (including SQLite support)
### What should be done?

We should document our new experimentation authentication support, including setup flags.

### Where?

Docs.

### Definition of Done

- Documentation updated.

### PR DESCRIPTION
**Pull request by Fashander**
# Description

Closes #3008.

## Readiness checklist

- [ ] I added/updated unit tests (and they pass).
- [ ] I added/updated integration/compatibility tests (and they pass).
- [ ] I added/updated comments and checked rendering.
- [ ] I made spot refactorings.
- [ ] I updated user documentation.
- [ ] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Milestone (`Next`), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

**Comment by Fashander** [comment]
@AlekSi  PTAL

File: website/docs/security/experimental-authentication-mode.md
Comment (AlekSi) [inline website/docs/security/experimental-authentication-mode.md:1]: Let's merge it into authentication.md under a separate section/header – we link to the main page in too many places

File: website/docs/security/authentication.md [hunk 1]
```diff
@@ -9,12 +9,18 @@ description: Learn to use authentication mechanisms
...
+
 For example, if the server was started with `postgres://user1:pass1@postgres:5432/ferretdb`,
 anonymous clients will be authenticated as user1,
 but clients that use `mongodb://user2:pass2@ferretdb:27018/ferretdb?tls=true&authMechanism=PLAIN` MongoDB URI will be authenticated as user2.
 Since usernames and passwords are transferred in plain text,
 the use of [TLS](../security/tls-connections.md) is highly recommended.
 
+:::tip
+FerretDB provides an experimental authentication mode that allows you to create user credentials for authenticated connections.
+See [experimental authentication mode](../security/authentication.md#experimental-authentication-mode) for more.
```
Comment (AlekSi) [inline website/docs/security/authentication.md:21]: That's the same file, no need to add file name

File: website/docs/security/authentication.md [hunk 2]
```diff
@@ -9,12 +9,18 @@ description: Learn to use authentication mechanisms
 FerretDB does not store authentication information (usernames and passwords) itself but uses the backend's authentication mechanisms.
 The default username and password can be specified in FerretDB's connection string,
 but the client could use a different user by providing a username and password in MongoDB URI.
+
```
Comment (AlekSi) [inline website/docs/security/authentication.md:12]: Do we need that change?

File: website/docs/security/authentication.md [hunk 3]
```diff
@@ -104,3 +110,102 @@ If you encounter any issues while authenticating with FerretDB, try setting the
 Please refer to your specific driver documentation on how to set this field.
 
 If this does not resolve your issue please file a bug report [here](https://github.com/FerretDB/FerretDB/issues/new?assignees=ferretdb-bot&labels=code%2Fbug%2Cnot+ready&projects=&template=bug.yml).
+
+## Experimental authentication mode
+
+FerretDB provides a new experimental authentication mode that supports the `SCRAM-SHA-1` and `SCRAM-SHA-256` authentication mechanisms.
+This mode enables FerretDB to manage users by itself and also provide support for more user management commands.
+
+You can enable this mode by setting the `FERRETDB_TEST_ENABLE_NEW_AUTH` flag or `--test-enable-new-auth` to `true`.
```
Comment (AlekSi) [inline website/docs/security/authentication.md:119]: We probably need to link to flags.md somewhere around there. We also should update flags.md (despite it saying "Do not document `--test-XXX` flags here" – that is an exception)

File: website/docs/security/authentication.md [hunk 4]
```diff
@@ -104,3 +110,102 @@ If you encounter any issues while authenticating with FerretDB, try setting the
...
+```
+
+With this new authentication mode, you can create user credentials for authenticated connections using the `createUser` command and also access other user management commands such as `dropAllUsersFromDatabase`, `dropUser`, `updateUser`, and `usersInfo`.
+
+This mode also enables you to set up initial authentication credentials for your instance.
+
+### Initial authentication setup
+
+You can secure your connections right from scratch by setting up an initial authentication credential using the following dedicated flags or environment variables.
+
```
Comment (AlekSi) [inline website/docs/security/authentication.md:134]: The same for flags.md

**Pull request merged**